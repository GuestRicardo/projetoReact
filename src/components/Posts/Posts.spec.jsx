import { Posts } from '.';
import { render, screen } from "@testing-library/react";

//mock para simular teste
const props = {
    posts: [
        {
            id: 1,
            title: 'titulo 1',
            body: 'Corpo 1',
            cover: 'img/img1.png',
        },
        {
            id: 2,
            title: 'titulo 2',
            body: 'Corpo 2',
            cover: 'img/img2.png',
        },
        {
            id: 3,
            title: 'titulo 3',
            body: 'Corpo 3',
            cover: 'img/img3.png',
        },
    ]
};
//fim do mock de teste

describe('<Posts />', () => {
    it('should render posts', () => {
        /**   const { debug } = */
        render(<Posts {...props} />);
        //debug();
        //heading
        expect(screen.getAllByRole('heading', { name: /title/i }))
            .toHaveLength(3);

        //imagem
        expect(screen.getAllByRole('img', { name: /title/i }))
            .toHaveLength(3);

        //imagem especifica 
        expect(screen.getAllByRole('img', { name: /title 3/i }))
            .toHaveAttribute('src', 'img/img3.png');

        //titulos paragrafos 
        expect(screen.getAllByText(/body/i))
            .toHaveLength(3);
    }); b
    //para testar quando nao tiver nenhum post
    it('should not render posts', () => {
        render(<Posts/>);
        expect(screen.queryByRole('heading', {name: /title/i}))
        .not.toBeInTheDocument();
    });
    it('should MATCH snapshot', () => {
        const { container } = render(<Posts {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

});