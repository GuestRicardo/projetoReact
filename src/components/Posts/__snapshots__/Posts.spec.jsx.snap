// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should MATCH snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="titulo 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        titulo 1
         
        1
      </h2>
      <p>
        Corpo 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="titulo 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        titulo 2
         
        2
      </h2>
      <p>
        Corpo 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="titulo 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        titulo 3
         
        3
      </h2>
      <p>
        Corpo 3
      </p>
    </div>
  </div>
</div>
`;
